js/src/launchAsync.ts

```diff
@@ -63,6 +74,259 @@ let launchResponseResolve: LaunchResponseResolve;
+                    break;
+                }
+                case FuncType.launchWithoutContentAsync: {
+                    resetTimeout(); // Reset the timeout once the reader page loads successfully. The Reader page will report further errors through PostMessage if there is an issue obtaining the ContentModel from the server
+                    readyForContent = true;
+                    const { sendContentIfReady } = extras;
+                    sendContentIfReady();
+                    break;
+                }
+            }
+        } else if (e.data === 'ImmersiveReader-Exit') {
+            exit(options.onExit);
+        } else if (e.data.startsWith(PostMessageLaunchResponse)) {
+            let launchResponse: LaunchResponse = null;
+            let error: Error = null;
+
+            let response: LaunchResponseMessage = null;
+            try {
+                response = JSON.parse(e.data.substring(PostMessageLaunchResponse.length));
+            } catch {
+                // No-op
+            }
+
+            if (response && response.success) {
+                const playMessageValue: any = {
+                    command: 'PlayState',
+                    parameters: 'Play'
+                };
+
+                const pauseMessageValue: any = {
+                    command: 'PlayState',
+                    parameters: 'Pause'
+                };
+
+                launchResponse = {
+                    container: iframeContainer,
+                    sessionId: response.sessionId,
+                    charactersProcessed: response.meteredContentSize,
+                    postLaunchOperations: {
+                        pause: () => {
+                            iframe.contentWindow!.postMessage(JSON.stringify({ messageType: 'InstrumentationCommand', messageValue: pauseMessageValue }), '*');
+                        },
+                        play: () => {
+                            iframe.contentWindow!.postMessage(JSON.stringify({ messageType: 'InstrumentationCommand', messageValue: playMessageValue }), '*');
+                        }
+                    }
+                };
+            } else if (response && !response.success) {
+                error = {
+                    code: response.errorCode,
+                    message: errorMessageMap[response.errorCode],
+                    sessionId: response.sessionId
+                };
+            } else {
+                error = {
+                    code: ErrorCode.ServerError,
+                    message: errorMessageMap[ErrorCode.ServerError]
+                };
+            }
+
+            isLoading = false;
+
+            if (funcType === FuncType.launchAsync) {
+                const {
+                    reject,
+                    resolve
+                } = extras;
+                if (launchResponse) {
+                    resetTimeout();
```

Comment:
Wasn't this function called already in this case?

Reply:
where? I dont believe so